pkg/scheduler/eventhandlers.go

```diff
@@ -21,23 +21,16 @@ import (
 	"reflect"
 	"strings"
 
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/client-go/dynamic/dynamicinformer"
-	"k8s.io/klog/v2"
-
 	v1 "k8s.io/api/core/v1"
 	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/dynamic/dynamicinformer"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/tools/cache"
-	v1helper "k8s.io/component-helpers/scheduling/corev1"
-	"k8s.io/component-helpers/scheduling/corev1/nodeaffinity"
-	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/kubelet/lifecycle"
```

Comment:
This doesn't look right. We should let kubelet depends on scheduler, not the other way around.

Generally speaking, we can have a shared function called `GeneralFilter(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo, includeAllFailures bool) ([]PredicateFailureReason, error)`.

- `includeAllReasons` should be false in preCheckForNode() as we want to fail fast
- `includeAllReasons` should be true for kubelet to return all failure reasons.

We can temporary `GeneralFilter` in this package, but the ultimate goal is to move it to "k8s.io/component-helpers/scheduling" - but that needs to tease apart other depencies such as NodeInfo as noderesources plugin.

Reply:
> but that needs to tease apart other depencies such as NodeInfo as noderesources plugin.

You are right. I think I will study this issue. When I have some ideas, can I open an issue to discuss it?    🐾